fv3core/utils/grid.py

```diff
@@ -371,6 +363,368 @@ def horizontal_starts_from_shape(self, shape):
     def grid_indexing(self) -> "GridIndexing":
         return GridIndexing.from_legacy_grid(self)
 
+    @property
+    def damping_coefficients(self) -> "DampingCoefficients":
+        return DampingCoefficients(
+            del6_u=self.del6_u,
+            del6_v=self.del6_v,
+            da_min=self.da_min,
+            da_min_c=self.da_min_c,
+        )
+
+    @property
+    def grid_data(self) -> "GridData":
+        horizontal = HorizontalGridData(
+            self.area,
+            self.rarea,
+            self.rarea_c,
+            self.dx,
+            self.dy,
+            self.dxc,
+            self.dyc,
+            self.dxa,
+            self.dya,
+            self.rdx,
+            self.rdy,
+            self.rdxc,
+            self.rdyc,
+            self.rdxa,
+            self.rdya,
+        )
+        vertical = VerticalGridData()
+        contravariant = ContravariantGridData(
+            self.cosa_u,
+            self.cosa_v,
+            self.cosa_s,
+            self.sina_u,
+            self.sina_v,
+            self.rsin_u,
+            self.rsin_v,
+            self.rsin2,
+        )
+        angle = AngleGridData(
+            self.sin_sg1,
+            self.sin_sg2,
+            self.sin_sg3,
+            self.sin_sg4,
+            self.cos_sg1,
+            self.cos_sg2,
+            self.cos_sg3,
+            self.cos_sg4,
+        )
+        return GridData(
+            horizontal_data=horizontal,
+            vertical_data=vertical,
+            contravariant_data=contravariant,
+            angle_data=angle,
+        )
+
+
+@dataclasses.dataclass(frozen=True)
+class HorizontalGridData:
+    """
+    Terms defining the horizontal grid.
+    """
+
+    area: FloatFieldIJ
+    rarea: FloatFieldIJ
+    # TODO: refactor this to "area_c" and invert where used
+    rarea_c: FloatFieldIJ
+    dx: FloatFieldIJ
+    dy: FloatFieldIJ
+    dxc: FloatFieldIJ
+    dyc: FloatFieldIJ
+    dxa: FloatFieldIJ
+    dya: FloatFieldIJ
+    # TODO: refactor usages to invert "normal" versions instead
+    rdx: FloatFieldIJ
+    rdy: FloatFieldIJ
+    rdxc: FloatFieldIJ
+    rdyc: FloatFieldIJ
+    rdxa: FloatFieldIJ
+    rdya: FloatFieldIJ
+
+    @property
+    def lon(self) -> FloatFieldIJ:
+        raise NotImplementedError()
+
+    @property
+    def lat(self) -> FloatFieldIJ:
+        raise NotImplementedError()
+
+
+@dataclasses.dataclass(frozen=True)
+class VerticalGridData:
+    """
+    Terms defining the vertical grid.
+
+    Eulerian vertical grid is defined by p = ak + bk * p_ref
+    """
+
+    # TODO: refactor so we can init with this,
+    # instead of taking it as an argument to DynamicalCore
+    # we'll need to initialize this class for the physics
+    @property
+    def ptop(self) -> float:
```

Comment:
I always thought this was a runtime constant, but makes sense to be defined in init when we can

Reply:
Yeah, I could imagine someone e.g. wanting to run a 2D shallow water model alongside a fully 3D model (e.g. to train a 2D ML model online), which would necessitate different ptop's. Even the physical constants might make sense to convert later for this kind of use case.